workbench-core/environments/src/postDeployment/cognitoSetup.ts

```diff
@@ -111,9 +105,12 @@ export default class CognitoSetup {
    * @returns user pool id
    */
   public async getUserPoolId(): Promise<string | undefined> {
-    const { STACK_NAME } = this._constants;
+    const { USER_POOL_NAME } = this._constants;
+    const poolNameParts = USER_POOL_NAME.split('-');
+    // We extract STACK_NAME (eg. swb-stage-region) from USER_POOL_NAME (eg. swb-userpool-stage-region)
+    const stackName = `${poolNameParts[0]}-${poolNameParts[2]}-${poolNameParts[3]}`;
     const describeStackParam = {
-      StackName: STACK_NAME
+      StackName: stackName
     };
 
     const stackDetails = await this._aws.clients.cloudformation.describeStacks(describeStackParam);
```

Comment:
in unit test, you can add a expect(mockCFN..describeStacks)toBeCalledWith(...) to check the describeStackParam is correct, so that you get a coverage of stackName extract correctly

Reply:
Added this check but in a different way